src/Sepes.Provisioning/Service/RoleProvisioningService.cs

```diff
@@ -104,22 +113,33 @@ async Task SetRoleAssignments(CloudResourceOperationDto operation, Study study,
+                desiredRoleAssignments = _participantRoleTranslatorService.CreateDesiredRolesForSandboxResourceGroup(study.StudyParticipants.ToList());
+                existingRoleAssignmentsForResource = await _azureRoleAssignmentService.GetResourceGroupRoleAssignments(operation.Resource.ResourceId, operation.Resource.ResourceName, cancellationToken);
             }
             else if (operation.Resource.Purpose == CloudResourcePurpose.StudySpecificDatasetContainer)
             {
-                desiredRoleAssignments = ParticipantRoleToAzureRoleTranslator.CreateDesiredRolesForStudyResourceGroup(study.StudyParticipants.ToList());
-            }               
-           
-            var existingRoleAssignments = await _azureRoleAssignmentService.GetResourceGroupRoleAssignments(operation.Resource.ResourceId, operation.Resource.ResourceName, cancellationToken);
+                desiredRoleAssignments = _participantRoleTranslatorService.CreateDesiredRolesForStudyDatasetResourceGroup(study.StudyParticipants.ToList());
+                existingRoleAssignmentsForResource = await _azureRoleAssignmentService.GetResourceGroupRoleAssignments(operation.Resource.ResourceId, operation.Resource.ResourceName, cancellationToken);
+            }
+            //Disabled for now. If roles must be differentiated for different datasets i same study, this will become relevant. As now now, roles are assigned on the resource group containing all dataset storage accounts
+            // else if (operation.Resource.Purpose == CloudResourcePurpose.StudySpecificDatasetStorageAccount)
+            // {
+            //     desiredRoleAssignments = TranslateParticipantToAzureRoleService.CreateDesiredRolesForStudyDatasetStorageAccountGroup(study.StudyParticipants.ToList());
+            //     existingRoleAssignmentsForResource = await _azureRoleAssignmentService.GetStorageAccountRoleAssignments(operation.Resource.ResourceId, operation.Resource.ResourceGroupName, operation.Resource.ResourceId, operation.Resource.ResourceName, cancellationToken);
+            // }     
+            else
+            {
+                throw new Exception($"Unable to determine role assignments, unknown purpose {operation.Resource.Purpose} for resource {operation.Resource.Id}");
+            }
             
             //Create desired roles that does not allready exist
             foreach (var curDesired in desiredRoleAssignments)
             {
-                var sameRoleFromExisting = existingRoleAssignments.Where(ra => ra.properties.principalId == curDesired.PrincipalId && ra.properties.roleDefinitionId.Contains(curDesired.RoleId)).FirstOrDefault();
+                var sameRoleFromExisting = existingRoleAssignmentsForResource.FirstOrDefault(ra => ra.properties.principalId == curDesired.PrincipalId && ra.properties.roleDefinitionId.Contains(curDesired.RoleId));
```

Comment:
Is there a difference here? Or is it the same, just written different?

Reply:
r is for resource, ra is for roleassignment, it's sort of given from what dbset one is querying. But i agree it might be a bit cryptic for developers who come in at a later state